### PR TITLE
HOPSFS-164: Override datanode addess with datanode loadbalance for K8…

### PR DESCRIPTION
**Pull request by smkniazi**
… deployments

### CODE REVIEW

File: internal/transfer/transfer.go [hunk 1]
```diff
@@ -99,13 +101,42 @@ func readBlockOpResponse(r io.Reader) (*hdfs.BlockOpResponseProto, error) {
...
 func getDatanodeAddress(datanode *hdfs.DatanodeIDProto, useHostname bool) string {
 	var host string
 	if useHostname {
 		host = datanode.GetHostName()
 	} else {
 		host = datanode.GetIpAddr()
 	}
+	port := datanode.GetXferPort()
+
+	//Uncomment for debugging
```
Comment (maismail) [inline internal/transfer/transfer.go:116]: maybe add a debug flag instead and only run this if debug

File: internal/transfer/transfer.go [hunk 2]
```diff
@@ -99,13 +101,42 @@ func readBlockOpResponse(r io.Reader) (*hdfs.BlockOpResponseProto, error) {
...
+	if portOverride != "" {
+		i, err := strconv.ParseInt(portOverride, 10, 32)
+		if err != nil {
+			fmt.Printf("Bad override port. %s:%s", HOPSFS_CLOUD_DATANODE_PORT_OVERRIDE_ENV, portOverride)
+			os.Exit(1)
+		}
+		port = uint32(i)
+	}
+
+	//Uncomment for debugging
```
Comment (maismail) [inline internal/transfer/transfer.go:135]: same as above

File: internal/transfer/transfer.go [hunk 3]
```diff
@@ -99,13 +101,42 @@ func readBlockOpResponse(r io.Reader) (*hdfs.BlockOpResponseProto, error) {
...
 	} else {
 		host = datanode.GetIpAddr()
 	}
+	port := datanode.GetXferPort()
+
+	//Uncomment for debugging
+	//oldAdd := fmt.Sprintf("%s:%d", host, port)
+
+	hostOverride := os.Getenv(HOPSFS_CLOUD_DATANODE_HOSTNAME_OVERRIDE_ENV)
+	portOverride := os.Getenv(HOPSFS_CLOUD_DATANODE_PORT_OVERRIDE_ENV)
```
Comment (maismail) [inline internal/transfer/transfer.go:120]: I think the PR is missing some commits, where are these overrides being used?

**Review by maismail — CHANGES_REQUESTED** [review]

File: internal/transfer/transfer.go (hunk 1 above)
Comment (smkniazi) [reply internal/transfer/transfer.go:116]: this lib does not have any mechanism for printing debug messages. I do not want to add a debug mechanism for this change.  I can add it later if needed.

**Review by maismail — APPROVED** [review]

**Pull request merged**